UI/log-redaction.cpp

```diff
@@ -0,0 +1,27 @@
+#include "log-redaction.hpp"
+
+using namespace std;
+
+string redact(const char *source)
+{
+	string login_replaced, mac_user_replaced, lin_user_replaced,
+		win_user_replaced, rtmps_replaced, srt_replaced,
+		connect_replaced;
+
+	login_replaced =
+		regex_replace(source, login_url_regex, "//user:pass@$1");
+	mac_user_replaced =
+		regex_replace(login_replaced, macos_user_regex, "/Users/user/");
+	lin_user_replaced = regex_replace(mac_user_replaced, linux_user_regex,
+					  "/home/user/");
+	win_user_replaced = regex_replace(lin_user_replaced, win_user_regex,
```

Comment:
Dunno how much of a difference it would realistically make, but you could probably ifdef the ones not applicable for the current operating system.